File: lab1/Rect.cpp
#include "Rect.h"

Rect::Rect(int w, int h) : Shape(0, 0), width(w), height(h) {}
Rect::Rect(int ix, int iy, int w, int h) : Shape(ix, iy), width(w), height(h) {}

int Rect::left() {
	return x;
}

int Rect::right() {
	return x + width;
}

int Rect::top() {
	return y;
}

int Rect::bottom() {
	return y + height;
}

int Rect::hypot2() {
	return width * width + height * height;
}

int Rect::maxSide() {
	return width > height ? width : height;
}

File: lab2/Models/Rect.cs
namespace ShapesDatabase.Models;

public class Rect : Shape {
    public int Width { get; set; }
    public int Height { get; set; }

    public override int left() => X;
    public override int right() => X + Width;
    public override int top() => Y;
    public override int bottom() => Y + Height;

    public int hypot2() => Width * Width + Height * Height;
    public int maxSide() => Width > Height ? Width : Height;

    public Rect(int width, int height) : base(0, 0) {
        Width = width;
        Height = height;
    }

    public Rect(int x, int y, int width, int height) : base(x, y) {
        Width = width;
        Height = height;
    }
}